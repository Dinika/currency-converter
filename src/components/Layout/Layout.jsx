import React from 'react';
import { Grid, Header } from 'semantic-ui-react';
import Aux from '../../HOC/Aux/Aux';
import TopBar from '../TopBar/TopBar';
import CurrencyConversionCard from '../../containers/CurrencyConversionCard/CurrencyConversionCard';
import Footer from '../Footer/Footer';
// Todo: Configure eslint to accept css loads

const Layout = () => (
  <Aux>
    <TopBar title="Currency Converter" />
    <Grid textAlign="center" verticalAlign="middle">
      <Grid.Column>
        <Header as="h4" textAlign="center">
          Convert Euros, Dollars, and Japanese Yen
        </Header>
        <CurrencyConversionCard />
      </Grid.Column>
    </Grid>
    <Footer title="Currency Converter" />
  </Aux>
);

export default Layout;
