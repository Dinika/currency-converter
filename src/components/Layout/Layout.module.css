.Grid {
    height: 100%;
}

.Column {
    max-width: 700px;
    height: 100%;
    display: flex !important;
    justify-content: center;
}

.Card {
    height: 50%;
    justify-content: center;
}