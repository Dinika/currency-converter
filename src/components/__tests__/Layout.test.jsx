import React from 'react';
import { shallow } from 'enzyme';
import Layout from '../Layout/Layout';

describe('Layout', () => {
  it('renders with all props set', () => {        
    const wrapper = shallow(
        <Layout />
    );
    expect(wrapper).toMatchSnapshot();
  });
});