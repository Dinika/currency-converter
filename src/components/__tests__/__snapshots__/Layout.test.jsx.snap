// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders with all props set 1`] = `
<Aux>
  <TopBar
    title="Currency Converter"
  />
  <Grid
    textAlign="center"
    verticalAlign="middle"
  >
    <GridColumn>
      <Header
        as="h4"
        textAlign="center"
      >
        Convert Euros, Dollars, and Japanese Yen
      </Header>
      <CurrencyConversionCard />
    </GridColumn>
  </Grid>
  <Footer
    title="Currency Converter"
  />
</Aux>
`;
